pyscf/adc/radc.py

```diff
@@ -1176,20 +1251,31 @@ def ip_adc_diag(adc,M_ij=None,eris=None):
 
     return diag
 
-
-def ea_contract_r_vvvv(myadc,r2,v_list):
+def ea_contract_r_vvvv(myadc,r2,vvvv):
 
     nocc = myadc._nocc
     nvir = myadc._nvir
 
     r2_vvvv = np.zeros((nocc,nvir,nvir))
     r2 = np.ascontiguousarray(r2.reshape(nocc,-1))
+    chnk_size = radc_ao2mo.calculate_chunk_size(myadc)
+
     a = 0
-    for dataset in v_list:
-         k = dataset.shape[0]
-         dataset = dataset[:].reshape(-1,nvir*nvir)
-         r2_vvvv[:,a:a+k] = np.dot(r2,dataset.T).reshape(nocc,-1,nvir)
-         a += k
+    if isinstance(vvvv, list):
+        for dataset in vvvv:
+             k = dataset.shape[0]
+             dataset = dataset[:].reshape(-1,nvir*nvir)
+             r2_vvvv[:,a:a+k] = np.dot(r2,dataset.T).reshape(nocc,-1,nvir)
+             del (dataset)
+             a += k
+    else :
```

Comment:
What about in-core implementation? Does in-core not fall under this else statement?